app/main.py

```diff
@@ -1 +1,26 @@
-# write your code here
+import json
+from decimal import Decimal
+
+
+def calculate_profit(json_file_name):
+    with open(json_file_name) as file:
+        file_content = json.load(file)
+
+    coins_amount = Decimal("0")
+    profit = Decimal("0")
+
+    for deal in file_content:
+        if deal["bought"]:
+            coins_amount += Decimal(deal["bought"])
+            profit -= Decimal(deal["bought"]) * Decimal(deal["matecoin_price"])
+        else:
```

Comment:
It would be better to use `if` here instead of `else`. Because theoretically there can be both 'sold' and 'bought'.